---
title: "DepMap"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
source("../differential_analysis/definitions_analysis_visualization.R")
```

# Load data

Depmap proteomics/ trancriptomics data

```{r}
depmap_trans <- readRDS("../r_data/depmap_transcriptomics_limma_wntgroup.RData")
depmap_prot <- readRDS("../r_data/depmap_proteomics_limma_wntgroup.RData")

depmap_reactome <- readRDS("../r_data/depmap_trans_prot_gsea_reactome.RData")
depmap_kegg <- readRDS("../r_data/depmap_trans_prot_gsea_kegg.RData")
depmap_cc <- readRDS("../r_data/depmap_trans_prot_gsea_cc.RData")
```

TCGA transcriptomics data

```{r}
high_low <- readRDS( "../r_data/DESeq2_res_high_low.RData")
high_norm <- readRDS("../r_data/DESeq2_res_high_norm.RData")
low_norm <- readRDS("../r_data/DESeq2_res_low_norm.RData")
```

```{r}
tcga_reactome <- readRDS( "../r_data/tcga_trans_gsea_reactome.RData")
tcga_cc <- readRDS( "../r_data//tcga_trans_gsea_cc.RData")
tcga_kegg <- readRDS("../r_data/tcga_trans_gsea_kegg.RData")
```

Write Supplementary Table
```{r}
library(xlsx)

cc_res_dep_rna <- depmap_reactome %>%
    dplyr::filter(evidence == "RNA") %>%
    dplyr::select(Description, setSize, enrichmentScore, pvalue, p.adjust)
kegg_res_dep_rna <- depmap_kegg %>%
    dplyr::filter(evidence == "RNA") %>%
    dplyr::select(Description, setSize, enrichmentScore, pvalue, p.adjust) 

cc_res_dep_prot <- depmap_reactome %>%
    dplyr::filter(evidence == "protein") %>%
    dplyr::select(Description, setSize, enrichmentScore, pvalue, p.adjust)
kegg_res_dep_prot <- depmap_kegg %>%
    dplyr::filter(evidence == "protein") %>%
    dplyr::select(Description, setSize, enrichmentScore, pvalue, p.adjust) 

cc_res_tcga_rna <- tcga_reactome %>%
    dplyr::filter(evidence == "RNA" & contrast == "high_low") %>%
    dplyr::select(Description, setSize, enrichmentScore, pvalue, p.adjust)
kegg_res_tcga_rna <- tcga_kegg %>%
    dplyr::filter(evidence == "RNA" & contrast == "high_low") %>%
    dplyr::select(Description, setSize, enrichmentScore, pvalue, p.adjust) 

cc_res_tcga_prot <- tcga_reactome %>%
    dplyr::filter(evidence == "protein" & contrast == "high_low") %>%
    dplyr::select(Description, setSize, enrichmentScore, pvalue, p.adjust)
kegg_res_tcga_prot <- tcga_kegg %>%
    dplyr::filter(evidence == "protein" & contrast == "high_low") %>%
    dplyr::select(Description, setSize, enrichmentScore, pvalue, p.adjust) 



cc_res <- merge(merge(cc_res_dep_rna, cc_res_dep_prot, by = "Description", suffix = c("_DepMap_RNA", "_DepMap_protein"), all = TRUE),
  merge(cc_res_tcga_rna, cc_res_tcga_prot, by = "Description", suffix = c("_TCGA_RNA", "_TCGA_protein"), all = TRUE),
  by = "Description", all = TRUE) %>%
  dplyr::mutate(number_hits = 4 - is.na(setSize_DepMap_RNA) -
      is.na(setSize_DepMap_protein) - 
      is.na(setSize_TCGA_RNA) -
      is.na(setSize_TCGA_protein)) %>%
  dplyr::filter(number_hits >1)

kegg_res <- merge(merge(kegg_res_dep_rna, kegg_res_dep_prot, by = "Description", suffix = c("_DepMap_RNA", "_DepMap_protein"), all = TRUE),
  merge(kegg_res_tcga_rna, kegg_res_tcga_prot, by = "Description", suffix = c("_TCGA_RNA", "_TCGA_protein"), all = TRUE),
  by = "Description", all = TRUE) %>%
  dplyr::mutate(number_hits = 4 - is.na(setSize_DepMap_RNA) -
      is.na(setSize_DepMap_protein) - 
      is.na(setSize_TCGA_RNA) -
      is.na(setSize_TCGA_protein)) %>%
  dplyr::filter(number_hits >1)

```





# Depmap transcriptomics/ proteomics

Metabolic pathways from KEGG modules
Mitochondrial things from CC and BP

```{r}

kegg_heatmap <- rbind(depmap_kegg %>% dplyr::mutate(dataset = "DepMap"),
      tcga_kegg %>% dplyr::filter(contrast == "high_low") %>% 
        dplyr::select(-contrast) %>%
        dplyr::mutate(dataset = "TCGA")) %>%
  group_by(Description) %>%
  dplyr::mutate(valid = n() >= 2) %>%
  dplyr::filter(valid) %>%
  dplyr::mutate(evidence = factor(evidence, levels = c("RNA", "protein"))) %>%
  ggplot(aes(x = evidence, y = Description, fill = enrichmentScore)) +
  geom_tile() + facet_wrap(~dataset) +
   scale_fill_gradient2(low = phm_low, mid = phm_mid, high = phm_high, midpoint = 0) +
  xlab("") + ylab("")
```

```{r}
# pathways displayed need to be reduced for figure readability
regex_pathways <- "respira|Mitochond|Cristae|Complex|metabolism"

reactome_heatmap <- rbind(depmap_reactome %>% dplyr::mutate(dataset = "DepMap"),
      tcga_reactome %>% dplyr::filter(contrast == "high_low") %>% 
        dplyr::select(-contrast) %>%
        dplyr::mutate(dataset = "TCGA")) %>%
  group_by(Description) %>%
  dplyr::mutate(valid = n() > 2) %>%
  dplyr::filter(valid) %>%
  dplyr::filter(grepl(regex_pathways, Description)) %>%
    dplyr::mutate(evidence = factor(evidence, levels = c("RNA", "protein"))) %>%
  ggplot(aes(x = evidence, y = Description, fill = enrichmentScore)) +
  geom_tile() + facet_wrap(~dataset) +
   scale_fill_gradient2(low = phm_low, mid = phm_mid, high = phm_high, midpoint = 0) +
  xlab("") + ylab("")
```

```{r}
pathways_interest
```



```{r}
kegg_pathways <- rbind(depmap_kegg %>% dplyr::mutate(dataset = "DepMap"),
      tcga_kegg %>% dplyr::filter(contrast == "high_low") %>% 
        dplyr::select(-contrast) %>%
        dplyr::mutate(dataset = "TCGA"))
```


```{r}

react_cc_pathways <-rbind(depmap_reactome %>% dplyr::mutate(dataset = "DepMap"),
      tcga_reactome %>% dplyr::filter(contrast == "high_low") %>%
    dplyr::select(-contrast) %>%
        dplyr::mutate(dataset = "TCGA"),
  depmap_cc %>% dplyr::mutate(dataset = "DepMap"),
  tcga_cc %>% dplyr::filter(contrast == "high_low") %>%
    dplyr::select(-contrast) %>%
        dplyr::mutate(dataset = "TCGA"))
```

```{r}
pathway_heatmap <- right_join(react_cc_pathways, 
pathways_interest, by = "Description") %>%
  dplyr::filter(!is.na(evidence) & Description != "Protein ubiquitination" & 
      Description != "Eukaryotic translation" & 
      Description != "Chromatin organization") %>%
    dplyr::mutate(evidence = factor(evidence, levels = c("RNA", "protein"))) %>%
  ggplot(aes(x = evidence, y = Description, fill = enrichmentScore)) +
  geom_tile() + 
  facet_grid(type~ dataset,
             space="free",
             scale="free_y")  +
   scale_fill_gradient2(low = phm_low, mid = phm_mid, high = phm_high, midpoint = 0) +
  xlab("") + ylab("") +
  theme_bw() + 
   theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
   theme(axis.text.x = element_text(angle = 45, hjust = 1),
  axis.text=element_text(size=7),
  axis.title=element_text(size=8),
  plot.title = element_text(size = 8),
     legend.title = element_text(size = 8),
       legend.text = element_text(size = 8)
     ) 
```

# TCGA transcriptomics/proteomics
```{r}
tcga_prot_res <- readRDS("../r_data/tcga_prot_res_high_low.RData")
```


# Dotplots
Glucolysis
TCA cycle

mitochondrial translation

```{r}
gene_ids <- clusterProfiler::bitr(high_low$SYMBOL, fromType = "SYMBOL", toType= c("SYMBOL", "ENTREZID"), OrgDb =org.Hs.eg.db) %>%
    dplyr::filter(!is.na(SYMBOL)) %>% unique()

library(reactome.db)
reactome_list <- as.list(reactomePATHID2EXTID)

# mito trna aminoacylation
mito_genes <- gene_ids %>% dplyr::filter(ENTREZID %in% c(reactome_list["R-HSA-379726"][[1]],
                                                              reactome_list["R-HSA-5368287"][[1]]))
```

Glycolysis R-HSA-70171
PYRUVATE METABOLISM  R-HSA-70268 part 1
```{r}
pyru1 <- c("ME1", "LDHA", "LDHB", "LDHC", "LDHAL6A", "SLC16A1", "SLC16A3", "SLC16A8", "BSG")
```

```{r}
glyco_genes <- gene_ids %>% dplyr::filter(ENTREZID %in% reactome_list["R-HSA-70171"][[1]] |
                                                SYMBOL %in% pyru1)
```

TCA cycle R-HSA-71403
```{r}
# pyru 2
pyru_2 <- gene_ids %>% dplyr::filter(ENTREZID %in% reactome_list["R-HSA-70268"][[1]]) %>%
  dplyr::filter(!(SYMBOL %in% pyru1))
```


```{r}
tca_genes <- rbind(gene_ids %>% dplyr::filter(ENTREZID %in% reactome_list["R-HSA-71403"][[1]]),
                   pyru_2)
```


```{r}
gene_selection <- c(mito_genes$SYMBOL, glyco_genes$SYMBOL, tca_genes$SYMBOL)

gene_list_complete <- data.frame(SYMBOL = gene_selection,
                        label = gene_selection,
                        pathway = c(rep("mitochondria", nrow(mito_genes)),
                                    rep("glycolysis", nrow(glyco_genes)),
                                    rep("TCA cycle", nrow(tca_genes)))) %>%
  dplyr::mutate(label = ifelse(label %in% genes_interest$label, label, NA))
```



```{r}
depmap_prot <- depmap_prot %>%
  dplyr::mutate(logFC = ifelse(abs(logFC)>5, sign(logFC)*5, logFC))

df0 <- left_join(depmap_prot, genes_interest, by = "SYMBOL") %>% 
  dplyr::filter(!is.na(pathway)) %>%
  dplyr::select(SYMBOL, logFC, adj.P.Val, pathway, label) %>%
  dplyr::mutate(contrast = "protein", dataset = "DepMap")

depmap_prot_volcano <- df0 %>%
  ggplot(aes(x = logFC, y = -log10(adj.P.Val))) +
    geom_point(data = depmap_prot, color = "grey", alpha = 0.4, size =0.8) +
  geom_point(aes(color = pathway), alpha = 0.4, size =1) +
  geom_text_repel(data  = subset(df0, logFC > 0),aes(label = label, color = pathway), xlim = c(2.5,5),
    show.legend=FALSE, size = 3) +
  geom_text_repel(data  = subset(df0, logFC < 0),aes(label = label, color = pathway), xlim = c(-5,-2.5), show.legend=FALSE, size =3) +
  geom_hline(yintercept = -log10(0.1)) + #facet_wrap(~pathway) +
  theme_bw() + theme(legend.position = "none") + ggtitle("DepMap proteomic")  +
  scale_color_manual(values = c(amino_c,glyco_c,mito_c,tca_c))
```
```{r}
depmap_trans <- depmap_trans %>%
  dplyr::mutate(logFC = ifelse(abs(logFC)>3, sign(logFC)*3, logFC)) %>%
  dplyr::mutate(adj.P.Val = ifelse(adj.P.Val < 0.00001, 0.00001, adj.P.Val))

df1 <- left_join(depmap_trans, genes_interest, by = "SYMBOL") %>% 
  dplyr::filter(!is.na(pathway)) %>%
  dplyr::select(SYMBOL, logFC, adj.P.Val, pathway, label) %>%
  dplyr::mutate(contrast = "RNA", dataset = "DepMap")

depmap_rna_volcano <- df1 %>%
  ggplot(aes(x = logFC, y = -log10(adj.P.Val))) +
    geom_point(data = depmap_trans, color = "grey", alpha = 0.4, size =0.8) +
  geom_point(aes(color = pathway), alpha = 0.4, size =1) +
  geom_text_repel(data  = subset(df1, logFC > 0),aes(label = label, color = pathway), xlim = c(1,2.5), show.legend=FALSE, size =3) +
  geom_text_repel(data  = subset(df1, logFC < 0),aes(label = label, color = pathway), xlim = c(-2.5,-1), show.legend=FALSE, size =3) +
  geom_hline(yintercept = -log10(0.1)) + #facet_wrap(~pathway) +
  theme_bw() + theme(legend.position = c(0.2,0.8),
    legend.title = element_text(size = 10),
  legend.text = element_text(size = 8),
  legend.key.size = unit(0.3, "cm"))  + ggtitle("DepMap transcriptomic") +
  scale_color_manual(values = c(amino_c,glyco_c,mito_c,tca_c))
```
```{r}
df2 <- left_join(tcga_prot_res, genes_interest, by = "SYMBOL") %>% 
  dplyr::filter(!is.na(pathway)) %>%
  dplyr::select(SYMBOL, logFC, adj.P.Val, pathway, label) %>%
  dplyr::mutate(contrast = "protein", dataset = "TCGA")

tcga_prot_volcano <- df2 %>%
  ggplot(aes(x = logFC, y = -log10(adj.P.Val))) +
    geom_point(data = tcga_prot_res, color = "grey", alpha = 0.4, size =0.8) +
  geom_point(aes(color = pathway), alpha = 0.4, size =1) +
  geom_text_repel(data  = subset(df2, logFC > 0),aes(label = label, color = pathway), xlim = c(1,2.5), show.legend=FALSE, size =3) +
  geom_text_repel(data  = subset(df2, logFC < 0),aes(label = label, color = pathway), xlim = c(-2.5,-1), show.legend=FALSE, size =3) +
  geom_hline(yintercept = -log10(0.1)) + #facet_wrap(~pathway) +
  theme_bw() + theme(legend.position = "none")  + ggtitle("TCGA proteomic")+
  scale_color_manual(values = c(amino_c,glyco_c,mito_c,tca_c))
```
```{r}
high_low1 <- high_low %>%
 # dplyr::mutate(log2FoldChange = ifelse(abs(log2FoldChange)>5, sign(log2FoldChange)*5, log2FoldChange)) %>%
  dplyr::mutate(padj = ifelse(padj < 10**-30,10**-30, padj))

df3 <- left_join(high_low1, genes_interest, by = "SYMBOL") %>% 
  dplyr::filter(!is.na(pathway)) %>%
  dplyr::rename(logFC = log2FoldChange,adj.P.Val = padj) %>%
  dplyr::select(SYMBOL, logFC, adj.P.Val, pathway, label) %>%
  dplyr::mutate(contrast = "RNA", dataset = "TCGA")

tcga_rna_volcano <- df3 %>%
  ggplot(aes(x = logFC, y = -log10(adj.P.Val))) +
    geom_point(data = high_low1, color = "grey", alpha = 0.4, size =0.8) +
  geom_point(aes(color = pathway), alpha = 0.4, size =1) +
  geom_text_repel(data  = subset(df3, logFC > 0),aes(label = label, color = pathway), xlim = c(1,2.5), show.legend=FALSE, size =3) +
  geom_text_repel(data  = subset(df3, logFC < 0),aes(label = label, color = pathway), xlim = c(-2.5,-1), show.legend=FALSE, size =3) +
  geom_hline(yintercept = -log10(0.1)) + #facet_wrap(~pathway) +
  theme_bw() + theme(legend.position = "none") + ggtitle("TCGA transcriptomic") +
  scale_color_manual(values = c(amino_c,glyco_c,mito_c,tca_c))
```

# Heatmap
```{r}
gene_heatmap <- rbind(rbind(df0, df1), rbind(df2, df3)) %>%
  ggplot(aes(x = contrast, y = SYMBOL, fill = logFC)) +
  geom_tile() + 
  scale_fill_gradient2(low = ghm_low, mid = ghm_mid, high = ghm_high, midpoint = 0) + 
  facet_grid(pathway ~ dataset,
             space="free",
             scales="free")  +
  xlab("") + ylab("") +
  theme_bw() + 
  scale_x_discrete(limits = c("RNA", "protein")) +
   theme( panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
   theme(axis.text.x = element_text(angle = 45, hjust = 1),
  axis.text=element_text(size=7),
  axis.title=element_text(size=8),
  plot.title = element_text(size = 8),
     legend.title = element_text(size = 8),
       legend.text = element_text(size = 8)
     ) 
```



Compare to healthy
AXIN2
ENO2 
PDHA1 
LARS2
SUCLG2

```{r}
sel_genes_ens <- clusterProfiler::bitr(c("AXIN2","ENO2" ,"PDHA1" ,"LARS2","SUCLG2", "PKLR"), fromType = "SYMBOL", toType= c("SYMBOL", "ENSEMBL"), OrgDb =org.Hs.eg.db) %>%
    dplyr::filter(!is.na(SYMBOL)) %>% unique()
```


```{r}
sampleTable <- readRDS("../r_data/TCGA_COAD_READ_rnaseq_annotation.RData")
# CMS class contains a "prediction" column that contains the predicted subtypes CMS1,2,3 or 4.
# if prediction was not successfull, this column is NA
CMS_class <- readRDS("../r_data/TCGA_CMS_classification_CMSclassifier.RData")

# RNAseq count data
all_counts <- readRDS( "../r_data/TCGA_allcounts_normalized.RData")
```
```{r}
counts_table <- all_counts %>% t() %>% as.data.frame() %>%
    rownames_to_column("id")
```

```{r}
annotated_counts <- left_join(left_join(counts_table, sampleTable , by = "id"), CMS_class, by = "id")
```

```{r}
annotated_counts  <- annotated_counts %>%
  dplyr::mutate(prediction = ifelse(sample_type != "Primary Tumor", "Normal tissue", prediction))
```

```{r}
plot_rna_cms <- function(gene_name, legend = FALSE) {
    name_conv <- high_low %>% dplyr::filter(SYMBOL == gene_name)
  name_conv$ENSEMBL
  
  dotplot <- annotated_counts %>%
  ggplot(aes_string(x = "ENSG00000168646", y = name_conv$ENSEMBL, color = "prediction")) +
  geom_point(size=0.8) + xlab("AXIN2") +
  scale_color_manual(values = c(cms1_c, cms2_c, cms3_c, cms4_c, "black"), na.value = gr3) +
  theme_bw() +
    scale_x_log10() + scale_y_log10() +
  theme(axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank())+ theme(legend.position = "top")
  
  if(!legend) {dotplot <- dotplot + theme(legend.position = "none")}

  violin <- annotated_counts %>%
    dplyr::filter(!is.na(prediction)) %>%
    ggplot(aes_string(x = "prediction", y = name_conv$ENSEMBL, fill = "prediction", color = "prediction")) +
    geom_violin() + ylab(gene_name) + scale_y_log10() +
    scale_color_manual(values = c(cms1_c, cms2_c, cms3_c, cms4_c, "black"), na.value = gr3) +
    scale_fill_manual(values = c(cms1_c, cms2_c, cms3_c, cms4_c, "black"), na.value = gr3) +
    theme_bw() + theme(legend.position = "none") +
    theme(axis.title.x=element_blank(),
          axis.text.x=element_blank(),
          axis.ticks.x=element_blank())

  plot_grid(violin, dotplot, rel_widths = c(0.7,1), align = "h")
}

```

LFC correlation
```{r}
high_n_fc <- left_join(high_norm, gene_list_complete, by = "SYMBOL") %>% dplyr::filter(!is.na(pathway)) %>% dplyr::select(SYMBOL, log2FoldChange, pathway, label)
low_n_fc <- left_join(low_norm, gene_list_complete, by = "SYMBOL") %>% dplyr::filter(!is.na(pathway)) %>% dplyr::select(SYMBOL, log2FoldChange)

pathway_vs_normal <- left_join(high_n_fc, low_n_fc, by ="SYMBOL", suffix = c(".high", ".low")) %>%
  ggplot(aes(x = log2FoldChange.low, y = log2FoldChange.high, color = pathway))  +
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0) +
  geom_abline(slope =1, color = "grey")+
  geom_point(size=1) +
  theme_bw() +
  ggtitle("TCGA transcriptomic, Tumor vs. normal") +
  scale_color_manual(values = c(glyco_c,tca_c,mito_c))+
  xlab("Wnt-low vs. normal") +
  ylab("Wnt-high vs. normal") 
```
Proteomics
```{r}

rnaseq <- read_delim("../external_data/Human__CPTAC_COAD__UNC__RNAseq__HiSeq_RNA__03_01_2017__BCM__Gene__BCM_RSEM_UpperQuartile_log2.cct", delim = "\t")

# this proteomics data is tumor vs. normal normalized
proteomics <- read_delim("../external_data/Human__CPTAC_COAD__PNNL__Proteome__TMT__03_01_2017__BCM__Gene__Tumor_Normal_log2FC.cct", delim = "\t")

annotation <- read_delim("../external_data/NIHMS1524432-supplement-8.csv", delim = "\t") %>%
    dplyr::filter(SampleID != "Type") %>%
    dplyr::rename(participant = SampleID)

axin2 <- rnaseq %>%
  dplyr::filter(attrib_name == "AXIN2")
wnt_group <- ifelse(axin2 > 10.5, "high", "low")[-1]

wnt_df <- data.frame(wnt_group = wnt_group,
           sample = colnames(axin2)[-1])

prot_df <- proteomics %>%
  pivot_longer(names_to = "sample", -attrib_name)
```
```{r}
prot_df <- left_join(prot_df, wnt_df, by = "sample") %>%
  drop_na() %>%
            dplyr::rename(SYMBOL = attrib_name)
```

```{r}
prot_sum <- prot_df %>% group_by(SYMBOL, wnt_group) %>%
  summarize(prot_abund = mean(value)) %>%
  pivot_wider(names_from = wnt_group, values_from = prot_abund)
```

```{r}
pathway_vs_normal_protein <- left_join(prot_sum, gene_list_complete , by ="SYMBOL") %>%
  dplyr::filter(!is.na(pathway)) %>%
  ggplot(aes(x = low, y = high, color = pathway)) +
  geom_hline(yintercept = 0) +
  geom_vline(xintercept = 0) +
  geom_abline(slope =1, color = "grey") +
  geom_point(size=1) +
 # ggrepel::geom_text_repel(aes(label = label)) + 
  theme_bw() +
  ggtitle("TCGA proteomic, Tumor vs. normal")+
  scale_color_manual(values = c(glyco_c,tca_c,mito_c)) +
  xlab("Wnt-low vs. normal") +
  ylab("Wnt-high vs. normal") 
```

Put RNAseq data into right format
```{r}
library(data.table)
rna_df <- rnaseq[,-1] %>% transpose()
colnames(rna_df) <- rnaseq$attrib_name
rna_df$participant <- colnames(rnaseq[-1])

rna_df <- rna_df %>%
    gather(key = "gene", value = "rna", -participant)

```

Put proteomics data into right format
```{r}
prot_df2 <- proteomics[,-1] %>% transpose()
colnames(prot_df2) <- proteomics$attrib_name
prot_df2$participant <- colnames(proteomics[-1])

prot_df2 <- prot_df2 %>%
    gather(key = "gene", value = "protein", -participant)
```

```{r}
plot_protein_level <- function(gene_name, legend = FALSE) {
  axin2_exp <- rna_df %>% dplyr::filter(gene == "AXIN2")
  cand_prot <- prot_df2 %>% dplyr::filter(gene == gene_name)

  dotplot <- inner_join(inner_join(axin2_exp, cand_prot, by = c("participant")),annotation,
           by = "participant") %>%
    dplyr::filter(!is.na(CMS)) %>%
    ggplot(aes(x = rna, y = protein, color = CMS)) +
    geom_point(size=0.8) +
    # the colors are assigned in the definitions.R file that I also use for the other figures
    scale_color_manual(values = c(cms1_c, cms2_c, cms3_c, cms4_c))+
    theme_bw() +
    ylab("Protein level") + xlab(expression(italic(AXIN2) ~ expression ~ (log[2]))) +
   # ggtitle("Human colon cancer\n(transcriptomic + proteomic)") +
    theme(
    axis.text=element_text(size=8),
    axis.title=element_text(size=10),
    plot.title = element_text(size = 10),
      axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank()
    ) + theme(legend.position = "top")
  
  if(!legend) {dotplot <- dotplot + theme(legend.position = "none")}
  
  violin <- inner_join(cand_prot, annotation,
           by = "participant") %>%
    dplyr::filter(!is.na(CMS)) %>%
    ggplot(aes(x = CMS, y = protein, fill = CMS, color = CMS)) +
    geom_violin() +
    # the colors are assigned in the definitions.R file that I also use for the other figures
    scale_fill_manual(values = c(cms1_c, cms2_c, cms3_c, cms4_c))+
    scale_color_manual(values = c(cms1_c, cms2_c, cms3_c, cms4_c))+
    theme_bw() +
      theme(legend.position = "none") +
    ylab(gene_name) +
    theme(axis.title.x=element_blank(),
          axis.text.x=element_blank(),
          axis.ticks.x=element_blank())

  plot_grid(violin, dotplot, rel_widths = c(0.5,1), align = "h")
}

```





```{r}
#volcanoes_1 <- plot_grid(depmap_rna_volcano, depmap_prot_volcano, ncol=1)
#volcanoes_2 <- plot_grid(tcga_rna_volcano, tcga_prot_volcano, ncol=1)
row1 <- plot_grid(pathway_heatmap + ggtitle("Gene set enrichment")+ 
    theme(plot.title = element_text(size = 12)), 
  gene_heatmap + ggtitle("Transcript and protein expression")+ 
    theme(plot.title = element_text(size = 12)), NULL,
  labels = c("A","B",""), nrow =1, rel_widths = c(1.1,0.7,0.4))

legend_plot <- annotated_counts %>%
  ggplot(aes_string(x = "ENSG00000168646", y = "ENSG00000168646", color = "prediction")) +
  geom_point(size=0.8) + xlab("AXIN2") +
  scale_color_manual(values = c(cms1_c, cms2_c, cms3_c, cms4_c, "black"), na.value = gr3) + theme_bw()

legend_b <- get_legend(
  legend_plot + 
    guides(color = guide_legend(nrow = 1)) +
    theme(legend.position = "bottom")
)

title1 <- ggdraw() + 
  draw_label(
    "Transcript expression",
    #fontface = 'bold',
    x = 0,
    hjust = -0.5
  ) +
  theme(
    # add margin on the left of the drawing canvas,
    # so title is aligned with left edge of first plot
    plot.margin = margin(0, 0, 0, 7)
  )

title2 <- ggdraw() + 
  draw_label(
    "Protein expression",
    #fontface = 'bold',
    x = 0,
    hjust = -0.5
  ) +
  theme(
    # add margin on the left of the drawing canvas,
    # so title is aligned with left edge of first plot
    plot.margin = margin(0, 0, 0, 7)
  )

row2_1 <- plot_grid(title1 , plot_rna_cms("HK1")  ,
  plot_rna_cms("MRPS31"), 
  plot_rna_cms("PDHA1"),
  plot_rna_cms("GOT2"), ncol=1, labels = c("C","","",""),
  rel_heights = c(0.2,1.2,1,1,1))

row2_2 <- plot_grid(title2, plot_protein_level("HK1") ,
  plot_protein_level("MRPS31"), 
  plot_protein_level("PDHA1"),
  plot_protein_level("GOT2"), ncol=1,labels = c("D","","",""),
  rel_heights = c(0.2,1.2,1,1,1))

dots_plots <- plot_grid(row2_1, row2_2, nrow=1)
legend_dot_plots <- plot_grid(dots_plots, legend_b, ncol=1, rel_heights = c(1,0.05))

row2_3 <- plot_grid(pathway_vs_normal, pathway_vs_normal_protein, NULL, labels = c("E","F", ""), ncol = 1)

row2 <- plot_grid(legend_dot_plots, row2_3, nrow=1, rel_widths = c(1.5,1))

fig6 <- plot_grid(row1,row2, ncol=1, rel_heights = c(1,1.7))

save_plot("assembled_figures/Figure6.pdf", fig6, base_height = 11, base_width = 10)

save_plot("assembled_figures/Figure6.png", fig6, base_height = 11, base_width = 10)
```

Suplementary Figure
```{r}
c1 <- plot_grid(title1 , plot_rna_cms("SLC16A1")  ,
  plot_rna_cms("PDK1"), ncol=1, labels = c("A","",""),
  rel_heights = c(0.2,1.2,1))

c2 <- plot_grid(title2, plot_protein_level("SLC16A1") ,
  plot_protein_level("PDK1"),ncol=1,labels = c("B","",""),
  rel_heights = c(0.2,1.2,1))

save_plot("../supplementary_material/FigureS5.pdf", plot_grid(c1,c2))
```

